### ISSUE TITLE
[$500] IOU-All categories are disabled but Car is displayed in distance request report

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#33147]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.13
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause-Internal Team
**Slack conversation:** @

## Action Performed:
Pre-condition: In admin account, disable all categories
1. Launch app
2. Tap employee WS chat
3. Tap plus icon -- request money --- Distance
4. Enter and finish waypoints
5. Tap next
6. Tap request
7. Tap on distance request created to open distance request details page
8. Tap category

## Expected Result:
When all categories are disabled in admin account and if user creates distance request without selecting any category, although default category, " car" must not be displayed.

## Actual Result:
When user creates distance request without selecting any category, default category " car" is displayed although all categories are disabled in admin account.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [x] Android: mWeb Chrome
- [ ] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/af7ab75c-b9f2-484a-aeac-780069f27ac0

![Bug6314252_1702632187015!img_20231215_104652](https://github.com/Expensify/App/assets/115492554/8423d25a-f4f0-4b38-9618-b2deca7d92c4)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~0195d4c5f6cd939683</li>
        <li>Upwork Job ID: 1735600591726432256</li>
        <li>Last Price Increase: 2023-12-15</li>
    </ul>
</details>

**Comment by ikevin127** [comment]
# Proposal

## Please re-state the problem that we are trying to solve in this issue

When user creates distance request without selecting any category, default category " car" is displayed although all categories are disabled in admin account.

## What is the root cause of that problem?

**TLDR:** We don't have logic that checks if any categories are enabled for a certain workspace when viewing a newly created distance request, which causes the preview Category option to be displayed with default back-end assigned category.

https://github.com/Expensify/App/blob/6c110a4c9b391c139d906e9a29561364aed6fcc7/src/libs/actions/IOU.js#L781

https://github.com/Expensify/App/blob/6c110a4c9b391c139d906e9a29561364aed6fcc7/src/pages/iou/request/step/IOURequestStepConfirmation.js#L172-L179

I applied some console logs inside the 2 functions mentioned above that come into action when the distance request is created and upon creating the request according to actions performed steps, both `category` and `transaction.category` are `undefined`.

<img width="529" alt="Screenshot 2023-12-15 at 17 41 38" src="https://github.com/Expensify/App/assets/56457735/c26c705b-b75e-4404-bd2d-494e7d75b287">

.
This means that the Car category is not assigned from the front-end when creating the request.

Even with all categories disabled from OD workspace admin dashboard, when creating the request in ND I believe the Car category is assigned by default on the back-end side when the request is created.

## What changes do you think we should make in order to solve the problem?

What we can do from the front-end side to fulfil the expected result is to hide the Category option in the request preview when the number of enabled categories is 0.

To do that we add an extra check in here:

https://github.com/Expensify/App/blob/3a3d6f32bc5dc5aad2f8355fc49d6a9ca5e3ae59/src/components/ReportActionItem/MoneyRequestView.js#L130

using `OptionsListUtils.getEnabledCategoriesCount(_.values(policyCategories))` like so:

```diff
+    const policyCategoriesCount = OptionsListUtils.getEnabledCategoriesCount(_.values(policyCategories));
+    const shouldShowCategory = isPolicyExpenseChat && (transactionCategory || OptionsListUtils.hasEnabledOptions(lodashValues(policyCategories))) && policyCategoriesCount > 0;

```

## Videos

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/56457735/a76c4a76-9bbb-41d8-a93b-42ef04496880

</details>

**Comment by shubham1206agra** [comment]
This is a backend issue, and this should be fixed in backend only. And we can apply @ikevin127 solution for disabling view of categories in Frontend after the BE fix and discussion in slack about this.

🎀👀🎀 C+ reviewed

**Comment by ikevin127** [comment]
Thanks for the review!

Given the advantage of decoupling in this particular case I think its worth considering implementing the front-end fix asap, independent of the back-end one instead of waiting for the back-end first - this way we cover both cases starting with the front-end.

These are the current options:
- wait for a back-end fix which would mean creating a distance request without a Category when none is selected or all of them are disabled (might take a while until it's prioritised)
- implement the front-end check proposed above which would fix the issue just like the back-end one would, plus having the advantage of being backwards compatible with a future back-end fix (can be done asap)

I checked from the pov of back-end creating the distance request without a Category and I can confirm that this issue will be fixed with the required back-end changes but I am not sure how good of an idea a back-end fix would be considering that it would mean creating distance requests without a default Category.

I'm looking forward to input from the Expensify team on this issue!

**Comment by DylanDylann** [comment]
@ikevin127 I think we only need to fix in BE sides because 
https://github.com/Expensify/App/blob/3a3d6f32bc5dc5aad2f8355fc49d6a9ca5e3ae59/src/components/ReportActionItem/MoneyRequestView.js#L130

In FE, we will hide the category if transactionCategory is empty or there is no enable category. Currently, `OptionsListUtils.hasEnabledOptions(lodashValues(policyCategories)` is enough in FE. Please correct me If I miss anything

**Comment by ikevin127** [comment]
I understand what you're saying.

> or there is no enable category

Currently when there's no enabled category (all disabled from OD) and we create a distance request (Category selector hidden in FE), the back-end assigns the Car category automatically even though the user did not select any Category because all of them are disabled from OD.

The debate is between:
- back-end changes that would allow creating a distance request without a Category when none chosen (because all are disabled)
- a one-liner front-end fix that checks if all categories are disabled and hides the request preview Category selector

**Comment by shubham1206agra** [comment]
> I understand what you're saying.
> 
> > or there is no enable category
> 
> Currently when there's no enabled category (all disabled from OD) and we create a distance request (Category selector hidden in FE), the back-end assigns the Car category automatically even though the user did not select any Category because all of them are disabled from OD.
> 
> The debate is between:
> 
>     * back-end changes that would allow creating a distance request without a Category when none chosen (because all are disabled)
> 
>     * a one-liner front-end fix that checks if all categories are disabled and hides the request preview Category selector

No, backend change is the only fix. Your FE change is just nice to have.

**Comment by shubham1206agra** [comment]
@yuwenmemon Can you take this internal?

**Comment by yuwenmemon** [comment]
@neil-marcellini confirming with you first that this is unexpected behavior? I think it makes sense to me that we shouldn't even show the category selector if all categories are disabled.

**Comment by yuwenmemon** [comment]
Asked about this here: https://expensify.slack.com/archives/C05DWUDHVK7/p1702928915733459

**Comment by yuwenmemon** [comment]
Backend fix is in review

**Comment by shubham1206agra** [comment]
@yuwenmemon Can we know the status of backend fix here?

**Comment by yuwenmemon** [comment]
Ah sorry, this is all fixed - closing!

**Comment by shubham1206agra** [comment]
@yuwenmemon Shouldn't we also apply https://github.com/Expensify/App/issues/33147#issuecomment-1858141488 fix in FE too?

**Comment by yuwenmemon** [comment]
@shubham1206agra We're going with a different solution here: https://github.com/Expensify/App/issues/33388
